extensions/amp-social-share/0.1/amp-social-share.js

```diff
@@ -69,106 +73,49 @@ class AmpSocialShare extends AMP.BaseElement {
     /** @private @const {number} */
     this.height_ = getLengthNumeral(this.element.getAttribute('height'));
 
-    this.renderShare_();
-  }
-
-  /**
-   * Renders the share based on the element config.
-   * @return {!Element}
-   */
-  renderShare_() {
-    const urlParams = {};
-
-    for (const param in this.typeConfig_['params']) {
-      const paramConf = this.typeConfig_['params'][param];
-      let paramValue = this.config_[param] || this.getDefaultValue_(
-          param, this.config_);
-      user.assert(!paramConf['required'] || paramValue !== null,
-          param + ' is a required attribute for ' + this.type_ + '. %s',
-        this.element);
-      if (paramValue == null && paramConf['type'] == 'fixed') {
-        paramValue = paramConf['value'];
-      }
-      if ('maxlength' in paramConf) {
-        const maxlength = paramConf['maxlength'];
-        user.assert(!paramValue || paramValue.length < maxlength,
-            param + ' cannot exceed ' + maxlength + '. %s', this.element);
-      }
-      if (paramValue != null) {
-        urlParams[paramConf['param']] = paramValue;
-      }
-    }
-
-    // Get the anchor or create one.
-    let link = elementByTag(this.element, 'a');
-    if (link == null) {
-      link = this.getWin().document.createElement('a');
-      link.textContent = this.typeConfig_['text'];
-      link.setAttribute('target', '_blank');
-
-      // Get the container or create one.
-      let container = elementByTag(this.element, 'span');
-      if (container == null) {
-        container = this.getWin().document.createElement('span');
-        container.classList.add(CLASSNAME_PREFIX + this.type_);
-
-        // Only add the container to the element if it didn't exist
-        this.element.appendChild(container);
-      }
-      container.appendChild(link);
+    /** @private @const {!Object} */
+    this.params_ = Object.assign({}, this.typeConfig_.defaultParams || {});
+    Object.assign(this.params_, getAttributesAsParams(
+        this.element, attr => {
+          const matches = attr.nodeName.match(/^data-param-(.+)/);
+          return (matches && matches[1]) || null;
+        }));
+
+    const hrefWithVars = addParamsToUrl(this.shareEndpoint_, this.params_);
+    urlReplacementsFor(this.getWin()).expand(hrefWithVars).then(href => {
+      this.element.setAttribute('href', href);
+    });
+    this.element.setAttribute('role', 'link');
+
+    // Install a global click listener only once for all amp-social-share.
```

Comment:
I don't have data but in general isn't adding lots of event listener affect the performance? If the user have 5 social share elements that's 5 event listeners. 

Not sure if it's needed, if you think using an event listener per element is fine I can do that as well, I think it'll be more readable as well.